src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -127,6 +127,8 @@ public override BoundNode VisitITuplePattern(BoundITuplePattern node)
                 case BoundDeclarationPattern _:
                 case BoundDiscardPattern _:
                 case BoundITuplePattern _:
+                case BoundTypePattern _:
```

Comment:
>case BoundTypePattern _: [](start = 16, length = 24)

shouldn't we learn that the input was non-null?

Reply:
No.  We only learn from pure null tests in this pass.

---
In reply to: [362991326](https://github.com/dotnet/roslyn/pull/40503#discussion_r362991326) [](ancestors = 362991326)